sherpa/tests/test_sherpa_io.py

```diff
@@ -0,0 +1,304 @@
+    """What happens if more columns than column names?"""
+
+    tfile = tmp_path / "col.dat"
+    tfile.write_text("# col2\n-0.2 23\n-0.4 24\n")
+
+    # Although ncols=1 is the default, be explicit in this test
+    (colnames, coldata, fname) = get_ascii_data(str(tfile), ncols=1)
+    assert colnames == ["col2"]
+    assert len(coldata) == 1
+    assert coldata[0] == pytest.approx([-0.2, -0.4])
+
+
+def test_get_ascii_data_more_columns_than_colnames_no_colkeys_ncols2(tmp_path):
+    """What happens if more columns than column names?"""
+
+    tfile = tmp_path / "col.dat"
+    tfile.write_text("# col2 col7\n-0.2 23 1\n-0.4 24 0\n")
+
+    (colnames, coldata, fname) = get_ascii_data(str(tfile), ncols=2)
+    assert colnames == ["col2", "col7"]
+    assert len(coldata) == 2
+    assert coldata[0] == pytest.approx([-0.2, -0.4])
+    assert coldata[1] == pytest.approx([23, 24])
+
+
+def test_get_ascii_data_more_columns_than_colnames_colkeys_ncols1(tmp_path):
+    """What happens if more columns than column names?
+
+    This also tests a mix-match between ncols and len(colkeys).
+    """
+
+    tfile = tmp_path / "col.dat"
+    tfile.write_text("# col2 col7\n-0.2 23 1\n-0.4 24 0\n")
+
+    # Although ncols=1 is the default, be explicit in this test
+    (colnames, coldata, fname) = get_ascii_data(str(tfile), ncols=1,
+                                                colkeys=["col7", "col2"])
+    assert colnames == ["col7", "col2"]
+    assert len(coldata) == 2
+    assert coldata[0] == pytest.approx([23, 24])
+    assert coldata[1] == pytest.approx([-0.2, -0.4])
+
+
+def test_get_ascii_data_more_columns_than_colnames_colkeys_ncols2(tmp_path):
+    """What happens if more columns than column names?
+
+    We need to set ncols=2 to pass the check for creating a
+    dstype=Data1D object (for some reason, when colkeys is not set,
+    then there is no check if ncols=1).
+
+    """
+
+    tfile = tmp_path / "col.dat"
+    tfile.write_text("# col2 col7\n-0.2 23 1\n-0.4 24 0\n")
+
+    (colnames, coldata, fname) = get_ascii_data(str(tfile), ncols=2,
+                                                colkeys=["col7", "col2"])
+    assert colnames == ["col7", "col2"]
+    assert len(coldata) == 2
+    assert coldata[0] == pytest.approx([23, 24])
+    assert coldata[1] == pytest.approx([-0.2, -0.4])
+
+
+def test_get_ascii_data_more_colnames_than_columns_no_colkeys_ncols1(tmp_path):
+    """What happens if more column names than columns: ncols=1?"""
+
+    tfile = tmp_path / "col.dat"
+    tfile.write_text("# col2 xcol ycol\n-0.2 23\n-0.4 24\n")
+
+    # Although ncols=1 is the default be explicit in this test
+    (colnames, coldata, fname) = get_ascii_data(str(tfile), ncols=1)
+    assert colnames == ["col2", "xcol", "ycol"]
+    assert len(coldata) == 1
+    assert coldata[0] == pytest.approx([-0.2, -0.4])
+
+
+def test_get_ascii_data_more_colnames_than_columns_no_colkeys_ncols2(tmp_path):
+    """What happens if more column names than columns: ncols=2?"""
```

Comment:
This behavior seems inconsistent with what happens if there are more columns than colnames, but I'm assuming it's not worth discussing because i will be handled in a later PR?

Reply:
Yeah - this is to try and ensure we have tests for the existing behaviour, so if we do change things later on we know about it. Or that's the plan; life has shown that you never catch all issues...